src/Analysis/Ast/Impl/Types/Collections/PythonCollectionType.cs

```diff
@@ -32,17 +32,17 @@ internal class PythonCollectionType : PythonTypeWrapper, IPythonCollectionType {
         /// </summary>
         /// <param name="typeName">Iterable type name. If null, name of the type id will be used.</param>
         /// <param name="collectionTypeId">Collection type id, such as <see cref="BuiltinTypeId.List"/>.</param>
-        /// <param name="interpreter">Python interpreter.</param>
+        /// <param name="declaringModule">Declaring module.</param>
         /// <param name="isMutable">Indicates if collection is mutable (like list) or immutable (like tuple).</param>
         public PythonCollectionType(
             string typeName,
             BuiltinTypeId collectionTypeId,
-            IPythonInterpreter interpreter,
+            IPythonModule declaringModule,
             bool isMutable
-        ) : base(collectionTypeId, interpreter.ModuleResolution.BuiltinsModule) {
+            ) : base(collectionTypeId, declaringModule) {
```

Comment:
Is this change related to databases or a separate bug fix?

Reply:
Yes and yes. Collections were assumed to be builtins like `list`, but `List[str]` is actually `typing` and restoration from the db didn't work since we ended up with `builtins:List[str]` that does not make sense.